http-client/Network/HTTP/Client/Types.hs

```diff
@@ -630,7 +661,17 @@ data Response body = Response
     --
     -- Since 0.1.0
     }
-    deriving (Show, Eq, T.Typeable, Functor, Data.Foldable.Foldable, Data.Traversable.Traversable)
+    deriving (Show, T.Typeable, Functor, Data.Foldable.Foldable, Data.Traversable.Traversable)
+
+instance Eq body => Eq (Response body) where
+  resp == resp' = and
+    [ responseStatus resp == responseStatus resp'
+    , responseVersion resp == responseVersion resp'
+    , responseHeaders resp == responseHeaders resp'
+    , responseBody resp == responseBody resp'
+    , responseCookieJar resp `equivCookieJar` responseCookieJar resp'  -- !
```

Comment:
I agree, this seems weird. I think it would be best to remove the instance for the first release of the major version. We can add it back later with changed behavior if there is demand for it.

Reply:
09862e36d4663c9a94d4ad6782b3873117a04864